Tests/MapboxMapsTests/Foundation/MapViewTests.swift

```diff
@@ -261,110 +261,112 @@ final class MapViewTests: XCTestCase {
+        attributionURLOpener = MockAttributionURLOpener()
+        mapView = MapView(
             frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 100)),
             mapInitOptions: MapInitOptions(),
             dependencyProvider: dependencyProvider,
             orientationProvider: orientationProvider,
             urlOpener: attributionURLOpener)
+        window = UIWindow()
         window.addSubview(mapView)
-        notificationCenter.removeObserverStub.reset()
-
-        mapView.removeFromSuperview()
-
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIApplication.didEnterBackgroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIApplication.willEnterForegroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIScene.didEnterBackgroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            return invocation.parameters.name == UIScene.willEnterForegroundNotification
-        }))
-        XCTAssertEqual(notificationCenter.removeObserverStub.invocations.count, 4)
-    }
 
-    func testDisplayLinkPausedWhenAppMovingToBackground() {
-        notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
-
-        XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [true])
+        metalView = try XCTUnwrap(dependencyProvider.makeMetalViewStub.invocations.first?.returnValue)
+        // reset is required here to ignore the setNeedsDisplay() invocation during initialization
+        metalView.setNeedsDisplayStub.reset()
     }
 
-    func testDisplayLinkResumedWhenAppMovingToForeground() {
-        notificationCenter.post(name: UIApplication.willEnterForegroundNotification, object: nil)
-
-        XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [false])
+    override func tearDown() {
+        metalView = nil
+        window = nil
+        mapView = nil
+        attributionURLOpener = nil
+        orientationProvider = nil
+        dependencyProvider = nil
+        locationProducer = nil
+        displayLink = nil
+        cameraAnimatorsRunnerEnablable = nil
+        bundle = nil
+        notificationCenter = nil
+        super.tearDown()
     }
 
     func testDisplayLinkResumedWhenSceneMovingToForeground() throws {
         guard #available(iOS 13.0, *) else {
             throw XCTSkip("Test requires iOS 13 or higher.")
         }
-        bundle.infoDictionaryStub.defaultReturnValue = ["UIApplicationSceneManifest": []]
-        mapView.didMoveToWindow()
 
+        mapView.didMoveToWindow()
```

Comment:
Does adding the view to the window in setUp call this for you?

Reply:
yes it should, it is better to have this call in setup.